osal/linux/osal.c

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed under the GNU General Public License version 2 with exceptions. See
+ * LICENSE file in the project root for full license information
+ */
+
+#include <time.h>
+#include <sys/time.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <string.h>
+#include <osal.h>
+#include "Python.h"
+
+#define USECS_PER_SEC     1000000
+
+int osal_usleep (uint32 usec)
+{
+   struct timespec ts;
+   ts.tv_sec = usec / USECS_PER_SEC;
+   ts.tv_nsec = (usec % USECS_PER_SEC) * 1000;
+   /* usleep is deprecated, use nanosleep instead */
+   int result;
+   Py_BEGIN_ALLOW_THREADS
+   int result = nanosleep(&ts, NULL);
+   Py_END_ALLOW_THREADS
```

Comment:
`Py_BEGIN_ALLOW_THREADS` and `Py_END_ALLOW_THREADS` are added to osal_usleep function to allow other threads to run when this piece of code is executed.